test/unit/server/BaseHttpServerFactory.test.ts

```diff
@@ -125,4 +125,75 @@ describe('A BaseHttpServerFactory', (): void => {
       expect(res.text).toBe(`${error.stack}\n`);
     });
   });
+
+  describe('A Base HttpServerFactory (With Unix Sockets)', (): void => {
+    const socket = getSocket('BaseHttpServerFactory');
+    const httpOptions = {
+      http: true,
+      showStackTrace: true,
+    };
+
+    afterAll(async(): Promise<void> => {
+      server.close();
+    });
+
+    beforeEach(async(): Promise<void> => {
+      jest.clearAllMocks();
+    });
+
+    if (!(process.platform === 'win32')) {
+      it('sends incoming requests to the handler.', async(): Promise<void> => {
+        const factory = new BaseHttpServerFactory(handler, httpOptions);
+        server = factory.startServer(socket);
+        await request(`http+unix://${socket.replace(/\//gui, '%2F')}`).get('/').set('Host', 'test.com').expect(200);
+
+        expect(handler.handleSafe).toHaveBeenCalledTimes(1);
+        expect(handler.handleSafe).toHaveBeenLastCalledWith({
+          request: expect.objectContaining({
+            headers: expect.objectContaining({ host: 'test.com' }),
+          }),
+          response: expect.objectContaining({}),
+        });
+      });
+
+      it('throws an error on windows.', async(): Promise<void> => {
+        const prevPlatform = process.platform;
+        Object.defineProperty(process, 'platform', {
+          value: 'win32',
+        });
+
+        const factory = new BaseHttpServerFactory(handler, httpOptions);
+        expect((): void => {
+          factory.startServer(socket);
+        }).toThrow();
+
+        Object.defineProperty(process, 'platform', {
+          value: prevPlatform,
+        });
+      });
+    } else {
+      it('throws an error when trying to start the server.', async(): Promise<void> => {
+        const factory = new BaseHttpServerFactory(handler, httpOptions);
+        expect((): void => {
+          factory.startServer(socket);
+        }).toThrow();
+      });
+
+      it('throws an error when trying to start the server on an unsupported platform.', async(): Promise<void> => {
```

Comment:
This test can be outside of the win32 if-statement right?

Reply:
To have 100% coverage on every platform there should be two tests on each platform, so on linux it should have one "succesful" test and one pretending to be on windows to see if it throws, and on windows it should have one "succesful" test where it throws and another where it does try to start up and then throws. Does that make sense?

Reply:
Sure, but if you move this test outside of the if-statement so it runs on both systems you should still have 100% coverage. Would just have to use a string that also fails on the linux machine.